terraform/aws/module-ec2/cloud-init.sh.tpl

```diff
@@ -0,0 +1,2 @@
+# cloud init conf passed as a variable 
```

Comment:
I would have removed this comment for a technical reason:

If the file start by `#cloud-config` it is read as a Cloud init yaml file see

```
#cloud-config
repo_update: true
repo_upgrade: all

packages:
 - httpd
 - mariadb-server
```
And if you don't put this header it is read like a shell script at least for Aws.

For this reason I would let the user fully control the content of the file